train_segmentation.py

```diff
@@ -540,7 +548,11 @@ def main(params, config_path):
 
     criterion = criterion.to(device)
 
-    filename = output_path.joinpath('checkpoint.pth.tar')
+    gdl_git_ref = get_key_def('gdl_git_reference', params['global'], None)
+    if gdl_git_ref is not None:
+        filename = output_path.joinpath(f'checkpoint_{Path(gdl_git_ref).name}.pth.tar')
```

Comment:
I propose to add the git tag or commit hash to the checkpoint name (taken from new gdl_git_reference config parameter when available). This would require also paying attention to the state_dict_path config at inference time.

Reply:
that's a good approach. I just suggested another approach in my PR #191. Take a look and let me know. In my PR, I don't ask user for git hash, I just getting from git command and store in hdf5 and checkpoint.tar.pth (after adding it to params)

Reply:
Your approach is probably better, the fewer things we have to enter manually in the config, the less likely we are to just leave it unchanged when making a new config. Let's get your solution through and I will revert my stuff accordingly.